### DISCUSSION TITLE
Python compatibility

### DISCUSSION THREAD

**Discussion by gleme** [dsoftwareinc/fakeredis-py#62] [Q&A] [answered]
Hey, I'm working on a project that is still implemented on top of `python2.7`. We are planning to migrate it to `python3` soon, but we are not so close to it yet. Currently, for unit tests, we basically use a spawned process of `redis-server` connected to a unix socket for each test case that depends on redis client, and for each one of these test cases, we spawn and kill this process when the test case context is ended, which takes a lot of time to run the tests, and sometimes it fails to execute some test cases due to race conditions (which for us is more critical). Also thinking about future steps, this setup would be very hard for us to run tests in parallel.

**TLDR**,  I didn't find it anywhere in the docs, so I'm wondering if the `fakeredis` package and the `lua` extension are `python 2.7` compatible? Because it would be good for us to implement a better test utils using `fakeredis` before we migrate to `python3`.

**Comment by cunla** [comment, marked as answer]
Hi,

`fakeredis` does not support python 2.7.
You can try to run the tests using python 2.7 but I doubt it will work.
